openedx/adg/admin_dashboard/performance/views.py

```diff
@@ -1,3 +1,151 @@
+from django.contrib.auth.models import User
+from django.db.models import Prefetch, Q
+from django.http import HttpResponse
+from django.views.generic import ListView, View
+from opaque_keys.edx.keys import CourseKey
+from six import StringIO
+
+from common.djangoapps.student.models import CourseEnrollment
+from lms.djangoapps.courseware.courses import get_course_by_id
+from openedx.adg.admin_dashboard.constants import DEFAULT_PAGINATE_BY
+from openedx.adg.admin_dashboard.performance.helpers import get_course_grades_for_users, get_course_grades_report_data
+from openedx.adg.lms.applications.models import BusinessLine, MultilingualCourseGroup
+from openedx.adg.lms.applications.rules import is_adg_admin
+
+
+class CourseGradesView(UserPassesTestMixin, ListView, View):
+    """
+    View to list and download grades of a course.
+    """
+
+    DEFAULT_COURSE_GRADES_PAGINATE_BY = DEFAULT_PAGINATE_BY
+    course = None
+    template_name = 'adg/admin_dashboard/performance/course_grades.html'
+    http_method_names = ('get', 'post')
+
+    def test_func(self):
+        """
+        Checks user permissions to access course grades.
+        Authenticated and staff user will be able to access course grades if he/she is a
+        superuser | adg admin | bu admin and course is a bu prerequisite for the same bu.
+
+        Returns:
+            bool: True if user has all permissions, False otherwise
+        """
+        user = self.request.user
+        self.course = self._get_course()
+
+        if not (user.is_authenticated and user.is_staff):
+            return False
+
+        if user.is_superuser or is_adg_admin(user):
+            return True
+
+        if not user.groups.filter(id__in=BusinessLine.objects.values_list('group__id')).exists():
+            return False
+
+        return MultilingualCourseGroup.objects.filter(
+            multilingual_courses__course__id=self.course.id
+        ).filter(
+            Q(is_common_business_line_prerequisite=True) | Q(business_line_prerequisite__group__in=user.groups.all())
+        ).exists()
+
+    def get_context_data(self, *, object_list=None, **kwargs):
+        """
+        Overridden to add course grades for the users.
+
+        Returns:
+            dict: Dict containing context data.
+        """
+        object_list = self.get_queryset().only('id', 'email', 'profile__name')
+        context_data = super().get_context_data(object_list=object_list)
+
+        user_ids = [user.id for user in context_data['object_list']]
+        course_grades = get_course_grades_for_users(user_ids, self.course.id)
+        for user in context_data['object_list']:
+            user.course_grade = course_grades[user.id]
+
+        return context_data
+
+    def get(self, request, *args, **kwargs):
+        """
+        Overridden to add extra context data.
+        """
+        self.extra_context = {
+            'course_name': self.course.display_name,
+        }
```

Comment:
Generally, with CBV, Django recommends that we [pass any extra context](https://docs.djangoproject.com/en/3.2/topics/class-based-views/generic-display/#adding-extra-context) by overriding the get_context_data implementation. Maybe we should move this there?

Reply:
Good catch, At first I avoided overriding the get_context_data so I used the `extra_context ` attribute to handle this. Now we are again going to remove get_context_data as there is no such need to override it. We can use this class attribute to add any extra context.

Reply:
I would still recommend that we add this extra context by overriding `get_context_data` as it is the usual way when working with CBV. But I leave this to your judgement.